lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -125,28 +125,17 @@ def get(self):
             # not in our API.
             if url.startswith(self.uri_prefix):
                 # If the URI is parameterized with a Flask "<type:name>"
-                # template string, we don't want to report it, so we remove
-                # it from the URI. We derive an API name by converting the
-                # "/" characters in the URI to "_", after removing a trailing
-                # "/" that would have been left by removing a template... we
-                # don't remove the trailing "/" from the URI, which serves as
-                # an indication that the parameter is needed.
+                # template string, we replace them with a simple 'x'.  We
+                # derive an API name by converting the "/" characters in the
+                # URI to "_".
                 #
                 # E.g, "/api/v1/controllers/list" yields:
                 #     "controllers_list": "/api/v1/controllers/list"
                 #
-                # while "/api/v1/users/<string:username>" yields:
-                #     "users": "/api/v1/users/"
-                #
-                # TODO: This won't work right with embedded template strings,
-                # which we're not currently using anywhere; but it'll require
-                # adjustment later if we add any. (E.g., something like
-                # "/api/v1/foo/<string:name>/detail/<string:param>")
-                url = self.param_template.sub("", url)
-                path = url[len(self.uri_prefix) + 1 :]
-                if path.endswith("/"):
-                    path = path[:-1]
-                path = path.replace("/", "_")
+                # while "/api/v1/foo/<string:username>/detail/<int:bar>" yields:
+                #     "users_x": "/api/v1/foo/x/detail/x"
+                url = self.param_template.sub("x", url)
+                path = url[len(self.uri_prefix) + 1 :].replace("/", "_")
```

Comment:
I'm OK with `/upload/ctrl/${controller}/${name}`, although I would pull for replacing `ctrl` with `controller`.

I don't mind the notion that there might be different kinds of "uploads" and that this URI is for controller-based tarballs.

With that as a base, I would regard everything after the `ctrl` (or `controller`...) to be a series of parameters.  As Dave indicated, we could have specified the request as `host://api/v1/upload/controller?controller=<str>&name=<str>`, but expressing it as `host://api/v1/upload/controller/<controller:str>/<filename:str>` is, I think, equivalent.  (And, I would be happy with either one.)

If we buy into _that_, then I think we're back at where we were with the original code:  we truncate the URI at the first template parameter, assume that everything after that point is just a series of one or more parameters, and render the API name as `upload_ctrl` (or `upload_controller`...).

I agree that this PR is not the right place for reworking this stuff.  But, if y'all agree with my way of thinking here, I think we can have the filename as part of the URI without having to worry about "embedded template strings", because they aren't really "embedded" -- they are just tacked on to the end with slash-separators.  :-)

Reply:
> I agree that this PR is not the right place for reworking this stuff. But, if y'all agree with my way of thinking here, I think we can have the filename as part of the URI without having to worry about "embedded template strings", because they aren't really "embedded" -- they are just tacked on to the end with slash-separators. :-)

It's still just kicking the problem down the road; but that's what I did before, and I'll accept doing it again. I would definitely prefer

```
    "users": "host://api/v1/upload/"
```

Over the current under-x forms.

Reply:
It's only kicking the problem down the road if the road is eventually leading to embedded parameters; otherwise, I think we have an adequate solution so long as we only have parameters which are "appended".

Reply:
Moved to #2287.